static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -0,0 +1,203 @@
+import {useState} from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import AvatarList from 'sentry/components/avatar/avatarList';
+import TeamAvatar from 'sentry/components/avatar/teamAvatar';
+import Badge from 'sentry/components/badge/badge';
+import Checkbox from 'sentry/components/checkbox';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import {CheckWrap} from 'sentry/components/compactSelect/styles';
+import UserBadge from 'sentry/components/idBadge/userBadge';
+import {InnerWrap, LeadingItems} from 'sentry/components/menuListItem';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t, tct} from 'sentry/locale';
+import type {Team} from 'sentry/types/organization';
+import type {User} from 'sentry/types/user';
+import {defined} from 'sentry/utils';
+import {useTeamsById} from 'sentry/utils/useTeamsById';
+import {useUser} from 'sentry/utils/useUser';
+import type {DashboardDetails, DashboardPermissions} from 'sentry/views/dashboards/types';
+
+interface EditAccessSelectorProps {
+  dashboard: DashboardDetails;
+  onChangeEditAccess?: (newDashboardPermissions: DashboardPermissions) => void;
+}
+
+/**
+ * Dropdown multiselect button to enable selective Dashboard editing access to
+ * specific users and teams
+ */
+function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorProps) {
+  const currentUser: User = useUser();
+  const dashboardCreator: User | undefined = dashboard.createdBy;
+  const {teams} = useTeamsById();
+  const teamIds: string[] = Object.values(teams).map(team => team.id);
+
+  let newDashboardPermissions = dashboard.permissions
+    ? structuredClone(dashboard.permissions)
+    : undefined;
+
+  const [selectedOptions, setselectedOptions] = useState<string[]>(
+    dashboard.permissions?.isCreatorOnlyEditable
+      ? ['_creator']
+      : ['_everyone', '_creator', ...teamIds]
+  );
+
+  let isEverythingSelected =
+    selectedOptions.length === ['_everyone', '_creator', ...teamIds].length;
```

Comment:
Makes sense. I removed the `isEverythingSelected` variable entirely, since it was essentially just being used to display the 'All' Badge in the dropdown button:
```
  // Avatars/Badges in the Edit Selector Button
  const triggerAvatars =
    selectedOptions.includes('_everyone') || !dashboardCreator ? (
      <StyledBadge key="_all" text={'All'} />
    ) : (
      <StyledAvatarList key="avatar-list" users={[dashboardCreator]} avatarSize={25} />
    );
```
(from [this file](https://github.com/getsentry/sentry/blob/c3444c2e0f3b2e469d00809e0f64b00f57e746e6/static/app/views/dashboards/editAccessSelector.tsx#L73))